src/main/java/org/elasticsearch/search/highlight/HighlightPhase.java

```diff
@@ -46,11 +45,19 @@
 public class HighlightPhase extends AbstractComponent implements FetchSubPhase {
 
     private final Highlighters highlighters;
+    private final ImmutableList<String> sortedHighlighters;
 
     @Inject
     public HighlightPhase(Settings settings, Highlighters highlighters) {
         super(settings);
         this.highlighters = highlighters;
+        // TODO: would be nice if each highlighter would have a precedence and we could just sort here by this value
+        // we could then also add custom highlighters to this list and they could be selected automatically depending on precendence
+        ImmutableList.Builder sortedListBuilder = ImmutableList.builder();
+        sortedListBuilder.add("fvh");
+        sortedListBuilder.add("postings");
+        sortedListBuilder.add("plain");
+        sortedHighlighters = sortedListBuilder.build();
```

Comment:
don't you have generic warnings here given that you don't specify the `<String>` type? I would maybe do `ImmutableList.<String>of("fvh","postings","plain")` ? why not static out of curiosity? I think it would be ok, list is going to be the same anyway on all nodes in the same jvm?